src/js/api/posts/create.mjs

```diff
@@ -0,0 +1,27 @@
+import { BASE_URL, optionsWithBody } from "../index.mjs";
+import {
+  renderSingleProfile,
+  renderUserSpecificPosts,
+} from "../../renders/index.mjs";
+import { errorMessage } from "../../templates/errorMessage.mjs";
+
+const createPostURL = `${BASE_URL}/posts`;
+
+/**
+ * Send the data to create a new post.
+ * Re-render the page after sending the data.
+ *
+ * @param {object} postData data input by the user
+ */
+export async function createPost(postData) {
+  try {
+    const options = optionsWithBody("POST", postData);
+    const response = await fetch(createPostURL, options);
+    const result = await response.json();
+    console.log(result);
+    renderUserSpecificPosts();
```

Comment:
Yes, this is exactly where I struggle to sort out. I did not like that event lister functions (is this what you mean narrative?) and API data functions became somehow interchangable despite my intention...
In this example renderUserSpecificPosts() is there to render a page after the form is sent, like the others. I tried to put it in eventListner but it did not update the page. I will try once again.

Reply:
Thank you for the message in Dicord. Got an idea and it is mind blowing! 
I would like to implement it, but it would take time to change the structure of core functions and test again. 
If I do not make it before the deadline, I will submit the current structure for now and play around in another branch.

Reply:
That's fine, it's more of a recommendation for next time than a change to make now